src/components/tabs/hooks/useTabPanel.ts

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+
+import {filterDOMProps} from '../../utils/filterDOMProps';
+import {bTabPanel} from '../constants';
+import {TabContext} from '../contexts/TabContext';
+import type {TabPanelProps} from '../types';
+
+export function useTabPanel(tabPanelProps: TabPanelProps) {
+    const tabContextProps = React.useContext(TabContext);
+
+    const currentValue = tabContextProps.value;
+    const parentId = tabContextProps.id;
+    const tabId = `${parentId}:t:${tabPanelProps.value}`;
+    const panelId = `${parentId}:p:${tabPanelProps.value}`;
+
+    const isSelected = currentValue === tabPanelProps.value;
+
+    return {
+        ...filterDOMProps(tabPanelProps, {labelable: true}),
```

Comment:
We will rewrite `aria-labelledby` below?

Reply:
I think we should change the order here

Reply:
Seems we must always rewrite this prop to ensure correct a11y here

Reply:
In this case users should not be able to set it

Reply:
changed